ipld.go

```diff
@@ -48,6 +47,36 @@ func (d Node) Context() interface{} {
 	return d[CtxKey]
 }
 
+// Like StripDirectivesAll but on the root node only, for use in Walk
+func (d Node) StripDirectives() Node {
```

Comment:
let's not have this on the `Node`, make it a separate function, like:

```go
func StripDirectives(n Node) Node { ... }
```

(want to keep the interface of `Node` very, very minimal.